File: ExpenseMaster/core/src/main/java/com/expensemaster/core/configuration/SpanConfig.java
package com.expensemaster.core.configuration;

import com.expensemaster.application.IApplicationSpan;
import com.expensemaster.core.ISpanAdapter;
import com.expensemaster.core.SpanAdapter;
import com.expensemaster.user.IUserSpan;
import io.opentelemetry.api.logs.LogRecordBuilder;
import io.opentelemetry.api.trace.Tracer;
import io.opentelemetry.context.propagation.TextMapPropagator;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Objects;

@Configuration
public class SpanConfig {
    private final Tracer tracer;
    private final TextMapPropagator textMapPropagator;
    private final LogRecordBuilder logRecordBuilder;

    public SpanConfig(
            final Tracer tracer,
            final TextMapPropagator textMapPropagator,
            final LogRecordBuilder logRecordBuilder
    ) {
        this.textMapPropagator = Objects.requireNonNull(textMapPropagator);
        this.tracer = Objects.requireNonNull(tracer);
        this.logRecordBuilder = Objects.requireNonNull(logRecordBuilder);
    }

    @Bean
    public ISpanAdapter createSpanAdapter() {
        return new SpanAdapter(tracer, textMapPropagator, logRecordBuilder);
    }
}

File: MailDispatcher/core/src/main/kotlin/com/maildispatcher/core/configuration/SpanConfig.kt
package com.maildispatcher.core.configuration

import com.maildispatcher.core.Span
import io.opentelemetry.api.trace.Tracer
import io.opentelemetry.context.propagation.TextMapPropagator
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class SpanConfig {
    @Bean
    open fun span(tracer: Tracer, textMapPropagator: TextMapPropagator): Span {
        return Span(tracer, textMapPropagator)
    }
}